hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -2025,7 +2251,8 @@ int SaraNcpClient::checkRunningImsi() {
     // Else, if not registered, check after CHECK_IMSI_TIMEOUT is expired
     if ((imsiCheckTime_ == 0) || ((connState_ != NcpConnectionState::CONNECTED) &&
             (millis() - imsiCheckTime_ >= CHECK_IMSI_TIMEOUT))) {
-        CHECK_PARSER(parser_.execCommand(UBLOX_CIMI_TIMEOUT, "AT+CIMI"));
+        // Do not error on timeout
```

Comment:
Because this is just for logs... and yeah it was noticed to cause a problem with initialization/registration when suffering through long registration.

Reply:
I understand this is just for logs, but normally a command should not timeout and not provide any response. Either we are not waiting long enough, or it's a R510 bug.

Reply:
There's a lot of reasons this can fail, more than just R510 bugs.  So I'd prefer not to increase the timeout beyond 10 seconds.  We've also already commented by the define that this should be an immediate response, and it's been observed to be 3 seconds or higher.  This is the section of code that just spits out the current IMSI for logging purposes, so if there's some error from this due to let's say the EtherSIM multi-imsi applet weirdness, we don't want it causing network teardowns and such.

Reply:
One thing I can think of here, when we initially added this code, it was to detect any multi IMSI weirdness and if it will be helpful for debugging purposes. There might be a chance that the SIM can go into a weird state for a brief period when scanning around, and in that case we likely don't want the ncp client to error and fall back? Since this is only for debugging purposes and it's causing problems, it looks OK not error on timeout.

Reply:
This is still buggy modem behavior if it doesn't respond and we should specifically be accounting for it instead of just silently ignoring and postponing any recovery for later potentially during an AT command with a very long timeout. `CHECK_PARSER()` (which sets `parserError_`) + `checkParser()` (which performs the recovery) is exactly the mechanism to recover from such cases. Let's leave `CHECK_PARSER()` in place and in the calling code make sure to call `checkParser()`, perhaps on errors.

Reply:
There are other modem calls though that will also perform this check for errors during registration if there is a larger problem at hand that is not just retrieval of the IMSI.  I would rather remove this command completely (since it's only for logging), than to allow the system to be torn down based on it erroring.  Again, I think it's just an edge case with R510 and Twilio Super SIMs, and shouldn't normally be erroring.  That said, I think having this command for logging is valuable, we should just not be too critical of it if it errors.  There are lots of other places in the code where we don't check for errors by design, and comment it as such.

Is there a way I can make a compromise and say... if we detect an error with AT+CIMI, immediately check for AT/OK, and if that's fine, do not report the error to parseError_ ?

Reply:
> if we detect an error with AT+CIMI, immediately check for AT/OK, and if that's fine

That's exactly what `CHECK_PARSER()` + `checkParser()` will do for you and why I proposed it :)

Reply:
@technobly Let's for now just wrap this call in `CHECK_PARSER()` and call it a day. We are not checking return value anyway from this function and the same problem exists with `interveneRegistration()`. At least all the other places where `checkParser()` is currently used will be aware that they might want to check that we get responses to AT commands. This should be okay for us, especially considering all the other mechanisms for checking modem responsiveness (muxer, ppp).

Reply:
I've got something coded and testing currently... all of the CHECK_ routines force a return, so they are not possible to follow up with another immediate check within the routine.  I know it's the same thing to wrap with CHECK_PARSER() and check for errors in the calling code, but I would like this to stay together self-contained with a nice comment to describe the full reasoning and logic.